File: 07b/main.py
from functools import cache

with open('input.txt', 'rt') as f:
    crabs = [int(x) for x in f.readline().split(',')]

@cache
def cost(distance):
    return sum(x for x in range(distance+1))

costs = [sum(cost(abs(c - p)) for c in crabs) for p in range(max(crabs))]
best_fuel = min(costs)
best_pos = costs.index(best_fuel)

print(best_fuel, best_pos)

File: 07b/ruby/main.rb
FILENAME = "input.txt"
f = File.open(FILENAME)
pos = f.read.split(",").map(&:to_i)

class Integer
  def cost
    acc = 0
    times {|i| acc += i}
    acc + self
  end
end

# This is lazy but it'll work...
# Figure out how much fuel it'll take to get to each position
min = 0
max = pos.max + 1
options = (min..max).map do |option|
  pos.map{|p| p-option}.map(&:abs).map(&:cost).sum
end

p options.min
